internal/server/handlers_accounts.go

```diff
@@ -72,7 +71,7 @@ func ListAccounts(env *HandlerConfig) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		accounts, err := env.DB.ListUsers()
 		if err != nil {
-			writeError(w, http.StatusInternalServerError, "Internal Error")
+			writeError(w, http.StatusInternalServerError, "Internal Error", env.Logger)
```

Comment:
I put the comment on this first line, but it is more general for the PR as a whole. We removed the logging calls of the specific errors when an error occurs and instead log them here, but we lose the specifics.

I would suggest adding an error parameter to `writeError` so that we can log those details. Otherwise, we would only log `Internal Error` here, with no details available to investigate the reason.

```suggestion
			writeError(w, http.StatusInternalServerError, "Internal Error", err, env.Logger)
```

The same comment would apply in most places, but particularly where we have an internal error.